sklearn/cluster/_optics.py

```diff
@@ -423,6 +423,22 @@ def _compute_core_distances_(X, neighbors, min_samples, working_memory):
     return core_distances
 
 
+@validate_params(
+    {
+        "X": [np.ndarray],
```

Comment:
The remaining error seems to mean that we support sparse matrix.
You should investigate if this is the case and correct the docstring then.

Reply:
Okay, thanks!
I will try to do it later